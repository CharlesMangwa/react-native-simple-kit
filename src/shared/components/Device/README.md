# Device ðŸ“±

A generic component to get access to the device's important infos.

### API

**No props needed**. If you want to change anything, you'll have to modify `src/components/Device/index.js` but also `<Device.Provider />` at the root of the project (`src/index.js`).

### Example

##### ./src/index.js

```jsx
/* @flow */
import React from 'react'
import { AppState, Dimensions, SafeAreaView, Text } from 'react-native'
import Device from '@components/Device'

const MyComponent = (): React$Element<*> => (
  <SafeAreaView>
    <Device.Provider
      value={{
        appState: AppState.currentState,
        isConnected: false,
        height: Dimensions.get('window').height,
        platform: Platform.OS,
        width: Dimensions.get('window').width,
      }}
    >
      <Navigation />
    </Device.Provider>
  </SafeAreaView>
)

export default MyComponent
```

#### somewhere_in_the_app.js

```jsx
/* @flow */
import React from 'react'
import { SafeAreaView, Text } from 'react-native'
import Device from '@components/Device'

const MyComponent = (): React$Element<*> => (
  <SafeAreaView>
    <Device.Consumer>
      {({ appState, height, isConnected, platform, width }) =>
        console.log(/* choose your weapon ðŸ˜Ž */)
      }
    </Device.Consumer>
  </SafeAreaView>
)

export default MyComponent
```

## Changelog

#### Friday, August 17th, 2018

- Initial release. We didn't track changes before this version.
