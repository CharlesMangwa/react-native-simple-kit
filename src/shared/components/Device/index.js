import { createContext } from 'react'
import { AppState, Dimensions, Platform } from 'react-native'

const Device = createContext({
  appState: AppState.currentState,
  height: Dimensions.get('window').height,
  isConnected: undefined,
  platform: Platform.OS,
  width: Dimensions.get('window').width,
})

export default Device
