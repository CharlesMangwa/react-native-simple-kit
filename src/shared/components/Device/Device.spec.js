import React from 'react'
import { View } from 'react-native'
import TestRenderer from 'react-test-renderer'
import Device from './index.js'

describe('ðŸ“± Generic component: <Device />', () => {
  it('should render `<Device.Provider />` correctly', () => {
    const component = TestRenderer.create(
      <Device.Provider>
        <View />
      </Device.Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render `<Device.Consumer />` correctly', () => {
    const component = TestRenderer.create(
      <Device.Consumer>{() => <View />}</Device.Consumer>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should pass correct data to `<Device.Consumer />`', () => {
    const expectedData = {
      appState: 'active',
      isConnected: false,
      height: 815,
      platform: 'ios',
      width: '375',
    }
    let receivedData = null
    TestRenderer.create(
      <Device.Provider value={expectedData}>
        <Device.Consumer>
          {props => {
            receivedData = props
            return null
          }}
        </Device.Consumer>
      </Device.Provider>
    )
    expect(receivedData).toMatchObject(expectedData)
  })
})
