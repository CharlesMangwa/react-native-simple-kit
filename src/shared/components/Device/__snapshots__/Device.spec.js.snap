// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ“± Generic component: <Device /> should render \`<Device.Consumer />\` correctly 1`] = `<View />`;

exports[`ðŸ“± Generic component: <Device /> should render \`<Device.Provider />\` correctly 1`] = `<View />`;
